protocol/daemons/liquidation/client/client.go

```diff
@@ -2,28 +2,43 @@ package client
 
 import (
 	"context"
+	"github.com/dydxprotocol/v4-chain/protocol/daemons/server/types"
+	timelib "github.com/dydxprotocol/v4-chain/protocol/lib/time"
 	"time"
 
-	"github.com/dydxprotocol/v4-chain/protocol/lib"
-
-	gometrics "github.com/armon/go-metrics"
 	"github.com/cometbft/cometbft/libs/log"
-	"github.com/cosmos/cosmos-sdk/telemetry"
-	"github.com/cosmos/cosmos-sdk/types/query"
 	appflags "github.com/dydxprotocol/v4-chain/protocol/app/flags"
 	"github.com/dydxprotocol/v4-chain/protocol/daemons/flags"
 	"github.com/dydxprotocol/v4-chain/protocol/daemons/liquidation/api"
 	daemontypes "github.com/dydxprotocol/v4-chain/protocol/daemons/types"
-	"github.com/dydxprotocol/v4-chain/protocol/lib/metrics"
 	clobtypes "github.com/dydxprotocol/v4-chain/protocol/x/clob/types"
 	satypes "github.com/dydxprotocol/v4-chain/protocol/x/subaccounts/types"
 )
 
+// Client implements a daemon service client that periodically queries a gRPC server for all liquidatable subaccounts
+// and reports them back to the protocol via the daemon server.
+type Client struct {
+	// Compose HealthCheckableImpl to implement the HealthCheckable interface.
+	daemontypes.HealthCheckable
+}
+
+// Ensure Client implements the HealthCheckable interface.
+var _ daemontypes.HealthCheckable = (*Client)(nil)
+
+func NewClient() *Client {
+	return &Client{
+		HealthCheckable: daemontypes.NewTimeBoundedHealthCheckable(
+			types.LiquidationsDaemonServiceName,
+			&timelib.TimeProviderImpl{},
+		),
+	}
+}
+
 // Start begins a job that periodically:
 // 1) Queries a gRPC server for all subaccounts including their open positions.
 // 2) Checks collateralization statuses of subaccounts with at least one open position.
 // 3) Sends a list of subaccount ids that potentially need to be liquidated to the application.
-func Start(
+func (c *Client) Start(
```

Comment:
Can we make this into `StartNewClient` like price daemon [here](https://github.com/dydxprotocol/v4-chain/blob/4bd3c34a0bf385e91bf5fb509115e19dcf9bd44a/protocol/daemons/pricefeed/client/client.go#L267)? And make `Start` private?

Reply:
`StartNewClient` itself is not necessarily the best design - we may wish to make the Start method blocking on the pricefeed daemon as well, and `StartNewClient` does not return any error result on start failure. I just didn't do this the last time I touched the pricefeed daemon's lifecycle in order to reduce scope and PR review-ability since I made so many changes.

Reply:
yup, let's stick to `NewClient` + `Start` paradigm. I believe you already created a tix to follow up [here](https://linear.app/dydx/issue/CORE-756/refactor-pricefeed-client-to-start-in-a-consistent-pattern-with-other)